src/boot/ThemeProvider.js

```diff
@@ -14,15 +15,22 @@ type Props = $ReadOnly<{|
   children: React$Node,
 |}>;
 
-class ThemeProvider extends PureComponent<Props> {
-  static defaultProps = {
+function ThemeProvider(props: Props) {
+  ThemeProvider.defaultProps = {
     theme: 'default',
   };
 
-  render() {
-    const { children, theme } = this.props;
-    return <ThemeContext.Provider value={themeData[theme]}>{children}</ThemeContext.Provider>;
+  const { children, theme } = props;
+  const colorScheme = useColorScheme();
+  let themePreferred = theme;
+
+  if (themePreferred === 'automatic') {
```

Comment:
I think `themeToUse` might be a bit better name than `themePreferred`, though I see that it's tricky to find the best names for things when we've already got `theme` and `colorScheme` floating around. 🤔

The reason is that this variable holds the answer (which we arrive at with help from `theme` and `colorScheme`) to the question, "So what theme should we actually use?". And, in particular, it doesn't represent the user's preference in the same sense that the user expresses a preference for light or dark when they press one of those items in the new Theme screen. (Because this variable might be light or dark as a result of following the OS-level setting.)